mergebot_poller.py

```diff
@@ -1,47 +1,133 @@
-"""TODO(jasonkuster): DO NOT SUBMIT without one-line documentation for mergebot_poller.
-
-TODO(jasonkuster): DO NOT SUBMIT without a detailed description of mergebot_poller.
+"""
+mergebot_poller is a utility which polls an SCM and follows commands.
 """
 
-import mergebot
+from github_helper import GithubHelper
+import merge
 import Queue
+from threading import Thread
+import time
+
+# TODO(jasonkuster): Fetch authorized users from somewhere official.
+AUTHORIZED_USERS = ["davorbonaci"]
+BOT_NAME = 'merge-bot'
+ORG = 'apache'
+
+
+class MergebotPoller(object):
+    """
+    MergebotPoller is a class which handles the polling of a configured SCM.
+    """
+
+    SCM_POLLERS = {'github': poll_github}
+    SCM_MERGERS = {'github': merge.merge_git}
+    CMD_MERGE = {'github': merge_git}
+    CMDS = {'merge': CMD_MERGE}
+
+    def __init__(self, config):
+        self.config = config
+        # Instantiate the two variables used for tracking work.
+        # Thread-safe because of the GIL.
+        self.work_queue = Queue.Queue()
+        self.known_work = set()
+        # Set up a github helper for handling network requests, etc.
+        self.github_helper = GithubHelper(ORG, self.config['repository'])
+
+    def poll(self):
+        """
+        Kicks off polling of the SCM. Reads SCM type from config and starts the
+        appropriate poller.
 
-GITHUB_SECRET = '../../github_auth/apache-beam.secret'
+        :return: returns nothing
```

Comment:
I don't think you need to say returns nothing -- the fact you don't document the return indicates that.